File: Whishpermate/WhisperMateShared/Models/Recording.swift
import Foundation

public struct Recording: Identifiable, Codable {
    public let id: UUID
    public let timestamp: Date
    public let transcription: String
    public let duration: TimeInterval?
    public let audioFileURL: URL?

    public init(id: UUID = UUID(), timestamp: Date = Date(), transcription: String, duration: TimeInterval? = nil, audioFileURL: URL? = nil) {
        self.id = id
        self.timestamp = timestamp
        self.transcription = transcription
        self.duration = duration
        self.audioFileURL = audioFileURL
    }

    public var formattedDate: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .medium
        formatter.timeStyle = .short
        return formatter.string(from: timestamp)
    }

    public var formattedDuration: String? {
        guard let duration = duration else { return nil }
        let minutes = Int(duration) / 60
        let seconds = Int(duration) % 60
        if minutes > 0 {
            return String(format: "%d:%02d", minutes, seconds)
        } else {
            return String(format: "%ds", seconds)
        }
    }
}

File: AIDictationAndroid/app/src/main/java/com/whispermate/aidictation/domain/model/Recording.kt
package com.whispermate.aidictation.domain.model

import java.time.Instant
import java.time.ZoneId
import java.time.format.DateTimeFormatter
import java.util.UUID

data class Recording(
    val id: String = UUID.randomUUID().toString(),
    val timestamp: Long = System.currentTimeMillis(),
    val transcription: String,
    val durationMs: Long? = null,
    val audioFilePath: String? = null
) {
    val formattedDate: String
        get() {
            val instant = Instant.ofEpochMilli(timestamp)
            val formatter = DateTimeFormatter.ofPattern("MMM d, yyyy 'at' h:mm a")
                .withZone(ZoneId.systemDefault())
            return formatter.format(instant)
        }

    val formattedDuration: String?
        get() = durationMs?.let { ms ->
            val seconds = ms / 1000
            val minutes = seconds / 60
            val remainingSeconds = seconds % 60
            "${minutes}:${remainingSeconds.toString().padStart(2, '0')}"
        }
}
